chips/msp432/src/uart.rs

```diff
@@ -0,0 +1,278 @@
+impl<'a> hil::uart::Uart<'a> for Uart<'a> {}
+
+impl<'a> hil::uart::Configure for Uart<'a> {
+    fn configure(&self, params: hil::uart::Parameters) -> ReturnCode {
+        // Disable module
+        let regs = self.registers;
+        regs.ctlw0.modify(usci::UCAxCTLW0::UCSWRST::SET);
+
+        // Setup module to UART mode
+        regs.ctlw0.modify(usci::UCAxCTLW0::UCMODE::UARTMode);
+
+        // Setup clock-source to SMCLK
+        regs.ctlw0.modify(usci::UCAxCTLW0::UCSSEL::SMCLK);
+
+        // Setup word-length
+        match params.width {
+            hil::uart::Width::Eight => regs.ctlw0.modify(usci::UCAxCTLW0::UC7BIT::CLEAR),
+            hil::uart::Width::Seven => regs.ctlw0.modify(usci::UCAxCTLW0::UC7BIT::SET),
+            hil::uart::Width::Six => {
+                panic!("UART: width of 6 bit is not supported by this hardware!")
+            }
+        }
+
+        // Setup stop bits
+        if params.stop_bits == hil::uart::StopBits::One {
+            regs.ctlw0.modify(usci::UCAxCTLW0::UCSPB::CLEAR);
+        } else {
+            regs.ctlw0.modify(usci::UCAxCTLW0::UCSPB::SET);
+        }
+
+        // Setup parity
+        if params.parity == hil::uart::Parity::None {
+            regs.ctlw0.modify(usci::UCAxCTLW0::UCPEN::CLEAR);
+        } else {
+            regs.ctlw0.modify(usci::UCAxCTLW0::UCPEN::SET);
+            if params.parity == hil::uart::Parity::Even {
+                regs.ctlw0.modify(usci::UCAxCTLW0::UCPAR::SET);
+            } else {
+                regs.ctlw0.modify(usci::UCAxCTLW0::UCPAR::CLEAR);
+            }
+        }
+
+        // Setup baudrate, all the calculation from the datasheet p. 915
+        let n = (self.clock_frequency / params.baud_rate) as u16;
+        let n_float = (self.clock_frequency as f32) / (params.baud_rate as f32);
+        let frac_part = n_float - (n as f32);
+        if n > 16 {
+            // Oversampling is enabled
+            regs.brw.set(n >> 4); // equals n / 16
+            let ucbrf = (((n_float / 16.0f32) - ((n >> 4) as f32)) * 16.0f32) as u16;
+            regs.mctlw
+                .modify(usci::UCAxMCTLW::UCBRF.val(ucbrf) + usci::UCAxMCTLW::UCOS16::SET);
+        } else {
+            // No oversampling
+            regs.brw.set(n);
+            regs.mctlw.modify(usci::UCAxMCTLW::UCOS16::CLEAR);
+        }
+
+        // Look for the closest calibration value
+        // According to the datasheet not the closest value should be taken but the next smaller one
+        let mut ucbrs = BAUD_FRACTIONS[0].reg_val;
+        for val in BAUD_FRACTIONS.iter() {
+            if val.frac > frac_part {
+                break;
+            }
+            ucbrs = val.reg_val;
+        }
+        regs.mctlw.modify(usci::UCAxMCTLW::UCBRS.val(ucbrs as u16));
+
+        // Enable module
+        regs.ctlw0.modify(usci::UCAxCTLW0::UCSWRST::CLEAR);
+
+        // Enable receive interrupt
+        // self.registers.ie.modify(usci::UCAxIE::UCRXIE::SET);
```

Comment:
Why isn't this enabled?

Reply:
Because the receive part of the UART isn't implemented yet. I didn't want to enable an interrupt when the whole handling logic is not implemented. My thought was that this would cause only problems.

Reply:
Ah ok. I think it's best just to remove this then. Otherwise looks good :)

Reply:
Ok, sure!